### PR TITLE
Add github actions for tf lint check

### PR DESCRIPTION
**Pull request by umeshkumhar**
- Add github actions for TF fmt, TF validate and TF init

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - uses: hashicorp/setup-terraform@v3
+        with:
+          terraform_version: "1.5.7"
+
+      - name: Terraform fmt
+        id: fmt
+        run: terraform fmt -check -recursive
+        continue-on-error: false
```
Comment (blackzlq) [inline .github/workflows/ci.yaml:21]: why we have continue-on-error only on fmt?

Comment (umeshkumhar) [reply .github/workflows/ci.yaml:21]: its not required though,, its default behavior to fail on error

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        run: terraform fmt -check -recursive
+
+      - name: Terraform Init
+        id: init
+        run: |
+          terraform -chdir=applications/rag init 
+          terraform -chdir=applications/ray init 
+          terraform -chdir=applications/jupyter init 
+
+      - name: Terraform Validate
```
Comment (andrewsykim) [inline .github/workflows/ci.yaml:29]: We already do this check in cloud build

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+jobs:
+  Terraform-Lint-Check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: hashicorp/setup-terraform@v3
+        with:
+          terraform_version: "1.5.7"
+
+      - name: Terraform fmt
```
Comment (andrewsykim) [inline .github/workflows/ci.yaml:18]: Can you just add this lint check to cloudbuild.yaml and remove the github actions workflow? I don't think we're at a point where we want to manage both cloudbuild and github actions for presubmit.

Comment (andrewsykim) [reply .github/workflows/ci.yaml:18]: (also because cloudbuild is a blocking pre-submit now)

**Pull request merged**